src/contradictory_claims/data/preprocess_cord.py

```diff
@@ -27,7 +34,147 @@ def filter_metadata_for_covid19(metadata_path: str, virus_lex_path: str, pub_dat
+                                                   'section': section}
+                                k = k + 1
+                            except KeyError:
+                                pass
+
+                            for temp_dict in temp_json['body_text'][0]:
+                                text = temp_dict['text']
+                                section = temp_dict['section']
+                                for key, v in replace_dict.items():
+                                    text = text.replace(key, v)
+                                    section = section.replace(key, v)
+                                covid19_dict[k] = {'cord_uid': cord_uid,
+                                                   'sentence': text,
+                                                   'section': section}
+                                k = k + 1
+
+#        if iter_num%100==0:
+#            print('Number of files read:', iter_num)
+
+    return pd.DataFrame.from_dict(covid19_dict, orient='index')
+
+
+def construct_regex_match_pattern(search_terms: List[str], search_type: bool = 0):
+    """
+    Construct regex search pattern for the specified terms.
+
+    :param terms_dict: list of search terms
+    :param search_type: 1 = exact pattern, 0 = fuzzy pattern
+    :return: Regex search pattern
+    """
+    if search_type == 1:
+        exact_pattern = '|'.join(search_terms)
+
+        return exact_pattern
+
+    else:
+        # TODO: fix flake8 error code FS001
+        fuzzy_terms = ['.*%s.*' % i for i in search_terms]  # noqa: FS001
+        fuzzy_pattern = '|'.join(fuzzy_terms)
+
+        return fuzzy_pattern
+
+
+def extract_regex_pattern(section_list: List[str], pattern: str):
+    """
+    Extract list of section names that match the specified regex pattern.
+
+    :param section_list: list of section names to search in
+    :param pattern: regex pattern to search for
+    :return: List of extracted section names
+    """
+    r = re.compile(pattern, re.IGNORECASE)
+    extracted_list = list(filter(r.match, section_list))
+    # remaining_list = list(set(section_list) - set(extracted_list))
+
+    return extracted_list
+
+
+def clean_text(input_data):
+    """
+    Filter text to keep only sentences containing at least 3 meaningful words.
+
+    :param input_data: pandas dataframe with publication text
+    :return: clean dataframe
+    """
+    # List of words to ignore
+    rep = {"text": "", "cite_spans": "", "ref_spans": "", "section": "", "abstract": "",
+           "biorxiv preprint": "", "medrxiv preprint": "", "doi:": ""}
+    rep = {re.escape(k): v for k, v in rep.items()}
+    pattern = re.compile("|".join(rep.keys()))
+    sentences_temp = [pattern.sub(lambda m: rep[re.escape(m.group(0))], s) for s in input_data.sentence.str.lower()]
+    pattern = re.compile(".*[A-Za-z].*")
+    sentences_to_keep = [(bool(re.search(pattern, s))) & (len(s.split(' ')) > 2) for s in sentences_temp]
```

Comment:
Ok this logic is again quite dense. Can this be broken up and commented more to add clarity?

Reply:
Done